GCC 10.0.1 build failure

Failed to build kiwix-lib with GCC 10.0.1 from Fedora 32:

```c++
[12/41] c++ -Isrc/25a6634@@kiwix@sha -Isrc -I../src -Iinclude -I../include -Istatic -I/usr/include/p11-kit-1 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -std=c++11 -O2 -g -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC -pthread -MD -MQ 'src/25a6634@@kiwix@sha/downloader.cpp.o' -MF 'src/25a6634@@kiwix@sha/downloader.cpp.o.d' -o 'src/25a6634@@kiwix@sha/downloader.cpp.o' -c ../src/downloader.cpp
FAILED: src/25a6634@@kiwix@sha/downloader.cpp.o 
c++ -Isrc/25a6634@@kiwix@sha -Isrc -I../src -Iinclude -I../include -Istatic -I/usr/include/p11-kit-1 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -std=c++11 -O2 -g -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC -pthread -MD -MQ 'src/25a6634@@kiwix@sha/downloader.cpp.o' -MF 'src/25a6634@@kiwix@sha/downloader.cpp.o.d' -o 'src/25a6634@@kiwix@sha/downloader.cpp.o' -c ../src/downloader.cpp
In file included from ../src/downloader.cpp:20:
../include/downloader.h:40:45: error: expected class-name before '{' token
   40 | class AriaError : public std::runtime_error {
      |                                             ^
../include/downloader.h: In constructor 'kiwix::AriaError::AriaError(const string&)':
../include/downloader.h:42:61: error: expected class-name before '(' token
   42 |   AriaError(const std::string& message) : std::runtime_error(message) {}
      |                                                             ^
../include/downloader.h:42:61: error: expected '{' before '(' token
```

Full build log: https://kojipkgs.fedoraproject.org//work/tasks/4395/41184395/build.log

Yes, it fixed issue, but some other cpp files need to be patched too:

```c++
../src/tools/regexTools.cpp: In function 'std::unique_ptr<icu_65::RegexMatcher> buildMatcher(const string&, icu_65::UnicodeString&)':
../src/tools/regexTools.cpp:39:17: error: 'out_of_range' in namespace 'std' does not name a type
   39 |   } catch (std::out_of_range&) {
      |                 ^~~~~~~~~~~~
../src/tools/regexTools.cpp:44:19: error: 'out_of_range' in namespace 'std' does not name a type
   44 |     } catch (std::out_of_range&) {
      |                   ^~~~~~~~~~~~
```

--- a/include/downloader.h
+++ b/include/downloader.h
@@ -25,6 +25,7 @@
 #include <map>
 #include <pthread.h>
 #include <memory>
+#include <stdexcept>
 
 namespace kiwix
 {

--- a/src/tools/regexTools.cpp
+++ b/src/tools/regexTools.cpp
@@ -25,6 +25,7 @@
 
 #include <memory>
 #include <map>
+#include <stdexcept>
 #include <pthread.h>
 
 std::map<std::string, std::shared_ptr<icu::RegexPattern>> regexCache;

